generators/generator-base.js

```diff
@@ -1787,6 +1789,64 @@ module.exports = class JHipsterBaseGenerator extends PrivateBase {
   }
 
   /**
+   * get a constraint name for tables in JHipster preferred style
+   *
+   * @param {string} entityName - name of the entity
+   * @param {string} columnOrRelationName - name of the column or related entity
+   * @param {string} prodDatabaseType - database type
+   * @param {boolean} noSnakeCase - do not convert names to snakecase
+   * @param {string} prefix - constraintName prefix for the constraintName
+   * @param {string} suffix - constraintName suffix for the constraintName
+   */
+  getConstraintName(entityName, columnOrRelationName, prodDatabaseType, noSnakeCase, prefix = '', suffix = '') {
+    let constraintName;
+    const legacyDbNames = this.jhipsterConfig && this.jhipsterConfig.legacyDbNames;
+    const separator = legacyDbNames ? '_' : '__';
+    if (noSnakeCase) {
+      constraintName = `${prefix}${entityName}${separator}${columnOrRelationName}${suffix}`;
+    } else {
+      constraintName = `${prefix}${this.getTableName(entityName)}${separator}${this.getTableName(columnOrRelationName)}${suffix}`;
+    }
+    let limit = 0;
+    // All versions of Oracle with a 30 character name limit have gone end-of-life, the limit is now 128
+    if (prodDatabaseType === ORACLE && constraintName.length > 128 && !this.skipCheckLengthOfIdentifier) {
+      this.warning(
+        `The generated constraint name "${constraintName}" is too long for Oracle (which has a 128 character limit). It will be truncated!`
+      );
+
+      limit = 128;
+    } else if (prodDatabaseType === MYSQL && constraintName.length > 64 && !this.skipCheckLengthOfIdentifier) {
+      this.warning(
+        `The generated constraint name "${constraintName}" is too long for MySQL (which has a 64 character limit). It will be truncated!`
+      );
+
+      limit = 64;
+    } else if (prodDatabaseType === POSTGRESQL && constraintName.length > 63 && !this.skipCheckLengthOfIdentifier) {
+      this.warning(
+        `The generated constraint name "${constraintName}" is too long for PostgreSQL (which has a 63 character limit). It will be truncated!`
+      );
+
+      limit = 63;
+    } else if (prodDatabaseType === MARIADB && constraintName.length > 64 && !this.skipCheckLengthOfIdentifier) {
+      this.warning(
+        `The generated constraint name "${constraintName}" is too long for MariaDB (which has a 64 character limit). It will be truncated!`
+      );
+
+      limit = 64;
+    }
+    return limit === 0
```

Comment:
I suggest to extract into `sql-constants.js`.
Would become really clean.
```
const databaseData = {
  [POSTGRESQL]: {
    name: 'PostgreSQL',
    constraintNameMaxLength: 63,
  },
  [MYSQL]: {
    name: 'MySql',
    constraintNameMaxLength: 64,
  },
  [MARIADB]: {
    name: 'MariaDB',
    constraintNameMaxLength: 64,
  },
  [ORACLE]: {
    name: 'Oracle',
    constraintNameMaxLength: 128,
  },
};

module.exports = {
  databaseData,
};
```

```suggestion
    const { name, constraintNameMaxLength } = databaseData[prodDatabaseType];
    if (constraintNameMaxLength && constraintName.length > constraintNameMaxLength && !this.skipCheckLengthOfIdentifier) {
      this.warning(
        `The generated constraint name "${constraintName}" is too long for ${name} (which has a ${constraintNameMaxLength} character limit). It will be truncated!`
      );
    }
    return !constraintNameMaxLength
```